### PR TITLE
Extract SectorAccessor from RetrievalProviderNode

### PR DESCRIPTION
**Pull request by dirkmc**

### CODE REVIEW

**Review by raulk — COMMENTED** [review]
This looks much cleaner. Tag me explicitly when you need a final review here, @dirkmc!

File: markets/retrievaladapter/provider.go [hunk 1]
```diff
@@ -52,42 +42,6 @@ func (rpn *retrievalProviderNode) GetMinerWorkerAddress(ctx context.Context, min
 	return mi.Worker, err
 }
 
-func (rpn *retrievalProviderNode) UnsealSector(ctx context.Context, sectorID abi.SectorNumber, offset abi.UnpaddedPieceSize, length abi.UnpaddedPieceSize) (io.ReadCloser, error) {
```
Comment (dirkmc) [inline markets/retrievaladapter/provider.go:55]: Just moved code to sectoraccessor.go

File: markets/retrievaladapter/provider.go [hunk 2]
```diff
@@ -104,29 +58,6 @@ func (rpn *retrievalProviderNode) GetChainHead(ctx context.Context) (shared.TipS
 	return head.Key().Bytes(), head.Height(), nil
 }
 
-func (rpn *retrievalProviderNode) IsUnsealed(ctx context.Context, sectorID abi.SectorNumber, offset abi.UnpaddedPieceSize, length abi.UnpaddedPieceSize) (bool, error) {
```
Comment (dirkmc) [inline markets/retrievaladapter/provider.go:107]: Just moved code to sectoraccessor.go

File: markets/retrievaladapter/provider.go [hunk 3]
```diff
@@ -175,37 +106,3 @@ func (rpn *retrievalProviderNode) GetRetrievalPricingInput(ctx context.Context,
 
 	return resp, nil
 }
-
-func (rpn *retrievalProviderNode) sectorsStatus(ctx context.Context, sid abi.SectorNumber, showOnChainInfo bool) (api.SectorInfo, error) {
```
Comment (dirkmc) [inline markets/retrievaladapter/provider.go:179]: Just moved code to sectoraccessor.go

File: markets/storageadapter/provider.go [hunk 4]
```diff
@@ -426,8 +422,4 @@ func (n *ProviderNodeAdapter) OnDealExpiredOrSlashed(ctx context.Context, dealID
 	return nil
 }
 
-func (n *ProviderNodeAdapter) IsUnsealed(ctx context.Context, sectorID abi.SectorNumber, offset abi.UnpaddedPieceSize, length abi.UnpaddedPieceSize) (bool, error) {
```
Comment (dirkmc) [inline markets/storageadapter/provider.go:429]: IsUnsealed is no longer needed (removed from the interface in markets as it was unused)

File: node/impl/client/client.go [hunk 5]
```diff
@@ -887,6 +899,17 @@ func (a *API) clientRetrieve(ctx context.Context, order api.RetrievalOrder, ref
...
+func (a *API) getTmpCarV2FilePath() (string, error) {
+	carsPath := filepath.Join(a.Repo.Path(), DefaultDAGStoreDir, "retrieval-cars")
+
+	if err := os.MkdirAll(carsPath, 0755); err != nil {
+		return "", xerrors.Errorf("failed to create dir")
+	}
+
+	return filepath.Join(carsPath, fmt.Sprintf("%d.car", time.Now().UnixNano())), nil
+}
+
```
Comment (dirkmc) [inline node/impl/client/client.go:912]: We're planning to replace this with a mechanism whereby caller can inject a blockstore, so just putting this temp code in as a stop-gap so everything compiles

**Pull request merged**